tempesta_fw/http.c

```diff
@@ -291,14 +292,14 @@ int
 tfw_http_send_404(TfwHttpReq *req)
 {
 	TfwStr rh = {
-		.ptr = (TfwStr []){
-			{ .ptr = S_404_PART_01, .len = SLEN(S_404_PART_01) },
-			{ .ptr = *this_cpu_ptr(&g_buf), .len = SLEN(S_V_DATE) },
-			{ .ptr = S_404_PART_02, .len = SLEN(S_404_PART_02) },
-			{ .ptr = S_CRLF, .len = SLEN(S_CRLF) },
+		.chunks = (struct TfwStr *)(TfwStr []){
+			{ .data = S_404_PART_01, .len = SLEN(S_404_PART_01) },
+			{ .data = *this_cpu_ptr(&g_buf), .len = SLEN(S_V_DATE) },
+			{ .data = S_404_PART_02, .len = SLEN(S_404_PART_02) },
+			{ .data = S_CRLF, .len = SLEN(S_CRLF) },
 		},
 		.len = SLEN(S_404_PART_01 S_V_DATE S_404_PART_02 S_CRLF),
-		.flags = 4 << TFW_STR_CN_SHIFT
+		.flags = 4
```

Comment:
Wrong. Please check all other similar places.
I am not commenting each and every one of them.